pkg/cmd/browse/browse.go

```diff
@@ -0,0 +1,250 @@
+	cmd.Flags().BoolVarP(&opts.WikiFlag, "wiki", "w", false, "Opens the wiki in browser")
+	cmd.Flags().BoolVarP(&opts.SettingsFlag, "settings", "s", false, "Opens the settings in browser")
+	cmd.Flags().BoolVarP(&opts.BranchFlag, "branch", "b", false, "Opens a branch in the browser")
+
+	return cmd
+}
+
+func openInBrowser(cmd *cobra.Command, opts *BrowseOptions) (exitCode, string) {
+
+	baseRepo, err := opts.BaseRepo()
+	httpClient, _ := opts.HttpClient()
+	apiClient := api.NewClientFromHTTP(httpClient)
+	branchName, err := api.RepoDefaultBranch(apiClient, baseRepo)
+
+	if !inRepo(err) { // must be in a repo to execute
+		printExit(exitNotInRepo, cmd, opts, "")
+		return exitNotInRepo, ""
+	}
+
+	if getFlagAmount(cmd) > 1 { // command can't have more than one flag
+		printExit(exitTooManyFlags, cmd, opts, "")
+		return exitTooManyFlags, ""
+	}
+
+	repoUrl := ghrepo.GenerateRepoURL(baseRepo, "")
+	response := exitSuccess
+
+	if !hasArg(opts) && hasFlag(cmd) {
+		response, repoUrl = addFlag(opts, repoUrl)
+	} else if hasArg(opts) && !hasFlag(cmd) {
+		response, repoUrl = addArg(opts, repoUrl, branchName)
+	} else if hasArg(opts) && hasFlag(cmd) {
+		response, repoUrl = addCombined(opts, repoUrl, branchName)
+	}
+
+	if response == exitSuccess {
+		opts.Browser.Browse(repoUrl) // otherwise open repo
+	}
+
+	printExit(response, cmd, opts, repoUrl) // print success
+	return response, repoUrl
+}
+
+func addCombined(opts *BrowseOptions, url string, branchName string) (exitCode, string) {
+
+	if !opts.BranchFlag { // gh browse --settings main.go
+		return exitInvalidCombo, ""
+	}
+
+	if opts.AdditionalArg == "" {
+		return exitSuccess, url + "/tree/" + opts.SelectorArg
+	}
+
+	arr := parseFileArg(opts)
+	if len(arr) > 1 {
+		return exitSuccess, url + "/tree/" + opts.AdditionalArg + "/" + arr[0] + "#L" + arr[1]
+	}
+
+	return exitSuccess, url + "/tree/" + opts.AdditionalArg + "/" + arr[0]
+
+}
+
+func addFlag(opts *BrowseOptions, url string) (exitCode, string) {
+	if opts.ProjectsFlag {
+		return exitSuccess, url + "/projects"
+	} else if opts.SettingsFlag {
+		return exitSuccess, url + "/settings"
+	} else if opts.WikiFlag {
+		return exitSuccess, url + "/wiki"
+	}
+	return exitExpectedArg, "" // Flag is a branch and needs an argument
+}
+
+func addArg(opts *BrowseOptions, url string, branchName string) (exitCode, string) {
+
+	if opts.AdditionalArg != "" {
+		return exitTooManyArgs, ""
+	}
+
+	if isNumber(opts.SelectorArg) {
+		url += "/issues/" + opts.SelectorArg
+		return exitSuccess, url
+	}
+
+	arr := parseFileArg(opts)
+	if len(arr) > 1 {
+		return exitSuccess, url + "/tree/" + branchName + "/" + arr[0] + "#L" + arr[1]
+	}
+
+	return exitSuccess, url + "/tree/" + branchName + "/" + arr[0]
+}
+
+func parseFileArg(opts *BrowseOptions) []string {
+	arr := strings.Split(opts.SelectorArg, ":")
+	return arr
+}
+
+func printExit(exit exitCode, cmd *cobra.Command, opts *BrowseOptions, url string) {
```

Comment:
As I've alluded to, the code in `cmd/gh/main.go` is responsible for printing errors. Your functions should return something like `fmt.Errorf("only a single flag is expected but, %d flags(s) were received", getFlagAmount(cmd))`,

The code in `cmd/gh/main.go` is responsible for printing the usage information.

For the success case, you can just directly issue a `fmt.Fprintln`.